lib/image.js

```diff
@@ -1,26 +1,89 @@
-'use strict';
```

Comment:
@LinusU until the memory issue is resolved, do you wanna pull out the nice refactoring you did in this file into a separate PR? Your version is way more readable 😍